src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -1025,10 +1025,8 @@ private static string[] GetLanguageIds()
         internal static PerformanceCounterLib GetPerformanceCounterLib(string machineName, CultureInfo culture)
         {
             string lcidString = culture.LCID.ToString("X3", CultureInfo.InvariantCulture);
-            if (machineName.CompareTo(".") == 0)
-                machineName = ComputerName.ToLower(CultureInfo.InvariantCulture);
-            else
-                machineName = machineName.ToLower(CultureInfo.InvariantCulture);
+
+            machineName = (machineName == "." ? ComputerName : machineName).ToLowerInvariant();
```

Comment:
Note: `string.CompareTo` does a culture-sensitive comparison based on the current culture. I don't think that was the intention here as everywhere else comparisons to `"."` are ordinal, so I've fixed it to do an ordinal comparison.